waddrmgr/manager.go

```diff
@@ -963,6 +972,152 @@ func (m *Manager) existsAddress(addressID []byte) (bool, error) {
 	return exists, nil
 }
 
+// ImportAddress imports the given address as a watch-only address into the address manager.
+//
+// All imported addresses will be part of the account defined by the
+// ImportedAddrAccount constant.
+func (m *Manager) ImportAddress(addr btcutil.Address, bs *BlockStamp) (ManagedAddress, error) {
+	m.mtx.Lock()
+	defer m.mtx.Unlock()
+
+	if !addr.IsForNet(m.chainParams) {
+		str := "address to be imported does not belong to the configured network"
+		return nil, managerError(ErrWrongNet, str, nil)
+	}
+
+	addressID := addr.ScriptAddress()
+	alreadyExists, err := m.existsAddress(addressID)
+	if err != nil {
+		return nil, err
+	}
+	if alreadyExists {
+		str := fmt.Sprintf("address for address ID %x already exists",
+			addressID)
+		return nil, managerError(ErrDuplicateAddress, str, nil)
+	}
+
+	// The start block needs to be updated when the newly imported address
+	// is before the current one.
+	updateStartBlock := bs.Height < m.syncState.startBlock.Height
+
+	var managedAddr ManagedAddress
+	switch addr := addr.(type) {
+	case *btcutil.AddressPubKeyHash:
+		// Encrypt public key hash.
+		pubKeyHash := addr.Hash160()[:]
+		encryptedPubKeyHash, err := m.cryptoKeyPub.Encrypt(pubKeyHash)
+		if err != nil {
+			str := fmt.Sprintf("failed to encrypt public key hash for %x",
+				addressID)
+			return nil, managerError(ErrCrypto, str, err)
+		}
+
+		// Create a new managed address based on the imported address.
+		managedAddr, err = newManagedPublicAddress(m, ImportedAddrAccount, pubKeyHash)
+		if err != nil {
+			return nil, err
+		}
+
+		// Save the new imported address to the db and update start block (if
+		// needed) in a single transaction.
+		err = m.namespace.Update(func(tx walletdb.Tx) error {
+			err := putImportedAddress(tx, addressID, ImportedAddrAccount,
+				ssNone, true, encryptedPubKeyHash, nil, nil)
+			if err != nil {
+				return err
+			}
+
+			if updateStartBlock {
+				return putStartBlock(tx, bs)
+			}
+
+			return nil
+		})
+	case *btcutil.AddressPubKey:
+		// Encrypt public key.
+		pubKey := addr.PubKey()
+		serializedPubKey := pubKey.SerializeCompressed()
```

Comment:
Thanks, let me update that. I think we can check whether the passed pubkey was compressed/uncompressed based on length and store in accordingly.